### PR TITLE
#10882 Slow Media Tab View

### PR DESCRIPTION
**Pull request by eraser2021999**
Signed-off-by: eraser2021999 <bcoticopol@gmail.com>

Fixed unresponsiveness of the Media Tab (Photo Gallery) - more fluent scrolling.
- implemented query search via created column and not modified to avoid cases when an older modified photo is display wrongly in the chronology;
- changed some logic in media loading to improve the loading speed of the gallery;

**Not fixed:**
- some issue with selecting a custom folder as gallery folder, I'm unsure if it worked correctly or not since the application crashed a lot;
- some UI issues with the gallery display rectangles;
- loading progress is not displayed correctly when scrolling faster, although the media is loading;

Manually tested with NextCloud Server (25.0.3) via simulator and hardware device (Google Pixel 6) with a media library of more than 10k items.

_edits by repo admin_
Issue links:
Partially solves #10425 (uses creation date instead of modification date)
Improves #10882

### CODE REVIEW

**Comment by eraser2021999** [comment]
Hi @tobiasKaminsky, @AlvaroBrey 

I believe I managed quite an improvement on the UI side, but I would suggest a thorough review and testing.

File: app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java [hunk 1]
```diff
@@ -146,52 +140,40 @@ private boolean parseMedia(long startDate, long endDate, List<Object> remoteFile
...
+            " new: " + filesAdded +
+            " updated: " + filesUpdated +
+            " deleted: " + filesDeleted +
+            " unchanged: " + unchangedFiles);
 
-    private boolean didNotFindNewResults(List<OCFile> filesToAdd,
-                                         List<OCFile> filesToUpdate,
-                                         Collection<OCFile> filesToRemove) {
-        return filesToAdd.isEmpty() && filesToUpdate.isEmpty() && filesToRemove.isEmpty();
+        return  (filesAdded + filesUpdated + filesDeleted + unchangedFiles > 0) ? false : true;
```
Comment (AlvaroBrey) [inline app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java:176]:
```suggestion
        final long totalFiles = filesAdded + filesUpdated + filesDeleted + unchangedFiles;
        return totalFiles <= 0;
```

Easier to read this way

**Review by AlvaroBrey — COMMENTED** [review]
Generally looks good, thanks @eraser2021999 

However, in order to merge this, please accept the suggestion below, and please fix DCO check, instructions here: https://github.com/nextcloud/android/pull/11281/checks?check_run_id=10787404887

File: app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java (hunk 1 above)
Comment (eraser2021999) [reply app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java:176]: Will do, thank you!

**Comment by eraser2021999** [comment]
It seems that my GitHub desktop broke something when tried to do the rebase --signoff. Pulling my hear out! :D

**Comment by AlvaroBrey** [comment]
Rebased and recreated the PR as #11288 to fix broken history here

**Pull request closed without merging**